applicaster-plugins/zapp-pipes-provider-promiboom
# Feature/getcollection handler

## Description: New feature

### Checklist for this pull request

- [x] PR is scoped to one task

### Related ticket

https://applicaster.atlassian.net/secure/RapidBoard.jspa?rapidView=191&modal=detail&selectedIssue=MS1-220&assignee=5d5ec0fb349a680d05e68899

### Detailed description

This PR creates the getCollection handler and the related code. Please, refer to the ticket for more detailed information.
The PR allows Zapp App Builder users to retrieve a collection of articles from the Promiboom API.
It's important to stress that no tests and error handling are included. Other tickets will take care of these improvements.
Also it's important to say that this is the first part of a wider integration of the Promiboom API features. Additional work is included in further tickets.
This PR includes also a description of the DSP in the README.md.
To test the DSP:

- Install the project
- Run it
- Make the following GET request:
http://localhost:8080/promiboom/fetchData?type=collection&collectionTitle=somethingfancy&location=home&text=dolor&count=1&offset=1
- Check that the retrieved data is of the following shape:
```{
    "type": {
        "value": "feed"
    },
    "title": "somethingfancy",
    "entry": [
        {
            "type": "article",
            "id": "36",
            "title": "Lorem Ipsum 3 with Image",
            "summary": "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua.",
            "published": "2019-09-03T11:36:52+00:00",
            "updated": "2019-09-19T16:50:03+00:00",
            "author": {
                "name": "Marius Günter"
            },
            "media_group": [
                {
                    "type": "IMAGE",
                    "media_item": [
                        {
                            "src": "http://promiboom.zweipro.de/files/2019/09/test1-600x314.jpg",
                            "type": "image",
                            "key": "small"
                        },
                        {
                            "src": "http://promiboom.zweipro.de/files/2019/09/test1-900x471.jpg",
                            "type": "image",
                            "key": "medium"
                        },
                        {
                            "src": "http://promiboom.zweipro.de/files/2019/09/test1-1200x628.jpg",
                            "type": "image",
                            "key": "large"
                        }
                    ]
                }
            ],
            "content": {
                "type": "html",
                "content": "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."
            }
        }
    ]
}
